### DISCUSSION TITLE
how to connect a contract to a contract runner with hardhat ethers v6?

### DISCUSSION THREAD

**Discussion by kuco23** [NomicFoundation/hardhat#4412] [Troubleshooting]
Hi, I want to connect a contract to an impersonated account on a forked network, using ethers v6. I followed a tutorial on how to do this, and I derived [this](https://github.com/kuco23/FAsset-Liquidator/blob/913719906890c92c9f098d47788a7ff6b8035237/test/liquidator-fork-coston.test.ts), but now (I'm guessing with v6 version) the `getSigner` method does not exist on ethers. I saw it exists on ethers provider object, but now I'm not sure how to sync it with hardhat's internal provider:
```typescript
import { ethers, network } from 'hardhat'
const provider = network.provider // hardhat network provider
```
I would appreciate any help with connecting a contract to the impersonated signer.

**Comment by fvictorio** [comment]
Your link 404s

**Reply by kuco23** [reply]
Sorry, have to learn to tag commits, fixed it.

**Reply by fvictorio** [reply]
`getSigner` exists in the `hre.ethers` object or, in your snippet example, the `ethers` object imported from Hardhat.
